src/incentives/ccip_bridge.sol

```diff
@@ -0,0 +1,190 @@
+        _;
+    }
+
+    modifier onlyWhitelistedSenders(address _sender) {
+        if (!whitelistedSenders[_sender]) revert SenderNotWhitelisted(_sender);
+        _;
+    }
+
+    modifier canBridge() {
+        require(!paused, "Bridging is paused");
+        _;
+    }
+
+    constructor(address router) CCIPReceiver(router) {
+        // Polygon
+        whitelistedDestinationChains[4051577828743386545] = true;
+        whitelistedSourceChains[4051577828743386545] = true;
+
+        // ETH
+        whitelistedDestinationChains[5009297550715157269] = true;
+        whitelistedSourceChains[5009297550715157269] = true;
+    }
+
+    fallback() external payable { }
+    receive() external payable { }
+
+    function setAffinePassAddress(address _affinePass) external onlyOwner {
+        affinePass = IAffinePass(_affinePass);
+    }
+
+    function setPaused(bool _paused) external onlyOwner {
+        paused = _paused;
+    }
+
+    function setChainReciever(uint64 _chainId, address _router) external onlyOwner {
+        chainReciever[_chainId] = _router;
+    }
+
+    function bridgePass(
+        uint64 destinationChainSelector,
+        address receiver,
+        uint256 id
+    ) external onlyWhitelistedDestinationChain(destinationChainSelector) canBridge returns (bytes32 messageId) {
+        require(affinePass.ownerOf(id) == msg.sender, "Not owner of token");
+
+        Client.EVM2AnyMessage memory message = _buildCCIPMessage(
+            chainReciever[destinationChainSelector],
+            receiver,
+            id,
+            address(0)
+        );
+
+        uint256 fee = IRouterClient(i_router).getFee(
+            destinationChainSelector,
+            message
+        );
+
+        if (fee > address(this).balance)
+            revert NotEnoughBalance(address(this).balance, fee);
+
+        messageId = IRouterClient(i_router).ccipSend{value: fee}(
+            destinationChainSelector,
+            message
+        );
+
+        emit BridgeRequest(messageId, destinationChainSelector, receiver, id);
+
+        affinePass.bridgeBurn(id);
+
+        return messageId;
+    }
+
+    function withdraw() public onlyOwner {
+        uint256 balance = address(this).balance;
+        payable(owner()).transfer(balance);
+    }
+
+    function whitelistDestinationChain(
+        uint64 _destinationChainSelector
+    ) external onlyOwner {
+        whitelistedDestinationChains[_destinationChainSelector] = true;
+    }
+
+    function denylistDestinationChain(
+        uint64 _destinationChainSelector
+    ) external onlyOwner {
+        whitelistedDestinationChains[_destinationChainSelector] = false;
+    }
+
+    function whitelistSourceChain(
+        uint64 _sourceChainSelector
+    ) external onlyOwner {
+        whitelistedSourceChains[_sourceChainSelector] = true;
+    }
+
+    function denylistSourceChain(
+        uint64 _sourceChainSelector
+    ) external onlyOwner {
+        whitelistedSourceChains[_sourceChainSelector] = false;
```

Comment:
Could this be reduced to a single function that sets the chain selector to a given bool?